RealmSwift/Realm.swift

```diff
@@ -435,8 +296,10 @@ public class Realm {
 
     :param: object The objects to be deleted.
     */
-    public func delete(objects: Results<Object>) {
-        rlmRealm.deleteObjects(objects)
+    public func delete<S: SequenceType where S.Generator.Element: Object>(objects: S) {
```

Comment:
That old results method did not work at all (nor does the current on which takes List). They need to be: 
```
    public func delete<T: Object>(objects: List<T>) {
        rlmRealm.deleteObjects(objects._rlmArray)
    }

    public func delete<T: Object>(objects: Results<T>) {
        rlmRealm.deleteObjects(objects.rlmResults)
    }
```